helpers/formatNumber.ts

```diff
@@ -1,13 +1,24 @@
 import { BigNumber } from "ethers";
-import { commify, formatEther } from "helpers";
+import { commify, formatEther, parseEther } from "helpers";
 
-export function formatNumberForDisplay(number: BigNumber | undefined, decimals = 2) {
+export function formatNumberForDisplay(
+  number: BigNumber | undefined,
+  options?: { decimals?: number; isFormatEther?: boolean }
+) {
   if (!number) return "0";
-  return truncateDecimals(commify(formatEther(number)), decimals);
+  const { decimals = 2, isFormatEther = true } = options || {};
+  const _number = isFormatEther ? formatEther(number) : number.toString();
+  return truncateDecimals(commify(_number), decimals);
 }
 
 export function truncateDecimals(number: string, decimals: number) {
   const [whole, decimal] = number.split(".");
   if (!decimal) return number;
+  if (decimals === 0) return whole.toString();
   return `${whole}.${decimal.slice(0, decimals)}`;
 }
+
+export function bigNumberFromFloatString(value: string) {
+  const truncated = truncateDecimals(value, 18);
```

Comment:
We need to truncate decimals in strings like this because if the number of decimal places exceeds the number of decimals for parse ether (18) then ethers will throw an error. These values are used for display purposes only.